tests/jobs/localisation/example_case/scripts/init_test_case.py

```diff
@@ -0,0 +1,262 @@
+def write_gen_obs(upscaled_values):
+    observation_dir = settings.observation.directory
+    obs_file_name = settings.observation.file_name
+    obs_data_dir = settings.observation.data_dir
+    cell_indx_list = settings.observation.selected_grid_cells
+    rel_err = settings.observation.rel_error
+    min_err = settings.observation.min_abs_error
+    if not os.path.exists(observation_dir):
+        print(f"Create directory: {observation_dir} ")
+        os.makedirs(observation_dir)
+    data_dir = observation_dir + "/" + obs_data_dir
+    if not os.path.exists(data_dir):
+        print(f"Create directory: {data_dir} ")
+        os.makedirs(data_dir)
+
+    print(f"Write observation file: {obs_file_name} ")
+    filename = observation_dir + "/" + obs_file_name
+    with open(filename, "w", encoding="utf8") as obs_file:
+        number = 0
+        for indices in cell_indx_list:
+            Iindx = indices[0] - 1
+            Jindx = indices[1] - 1
+            Kindx = indices[2] - 1
+
+            value = upscaled_values[Iindx, Jindx, Kindx]
+            value_err = math.fabs(value) * rel_err
+            value_err = max(value_err, min_err)
+
+            obs_data_relative_file_name = (
+                obs_data_dir
+                + "/obs_"
+                + str(Iindx + 1)
+                + "_"
+                + str(Jindx + 1)
+                + "_"
+                + str(Kindx + 1)
+                + ".txt"
+            )
+
+            obs_file.write(f"GENERAL_OBSERVATION   OBS_{Iindx+1}_{Jindx+1}_{Kindx+1}  ")
+            obs_file.write("{ ")
+            obs_file.write(
+                f"DATA = RESULT_UPSCALED_FIELD ; INDEX_LIST = {number} ; RESTART = 0;  "
+            )
+            obs_file.write(f"OBS_FILE = ./{obs_data_relative_file_name} ; ")
+            obs_file.write(" };\n")
+            number += 1
+            data_file_name = observation_dir + "/" + obs_data_relative_file_name
+            print(f"Write file: {data_file_name} ")
+            with open(data_file_name, "w", encoding="utf8") as data_file:
+                data_file.write(f"{value}  {value_err}\n")
+
+
+def create_grid():
+    grid_file_name = settings.field.grid_file_name
+    nx, ny, nz = settings.field.grid_dimension
+    xsize = settings.grid_size.xsize
+    ysize = settings.grid_size.ysize
+    zsize = settings.grid_size.zsize
+    if settings.grid_size.use_eclipse_grid_index_origo:
+        flip = -1
+        x0 = 0.0
+        y0 = ysize
+        z0 = 0.0
+    else:
+        flip = 1
+        x0 = 0.0
+        y0 = 0.0
+        z0 = 0.0
+
+    dx = xsize / nx
+    dy = ysize / ny
+    dz = zsize / nz
+
+    grid_object = xtgeo.create_box_grid(
+        dimension=(nx, ny, nz),
+        origin=(x0, y0, z0),
+        increment=(dx, dy, dz),
+        rotation=0.0,
+        flip=flip,
+    )
+
+    print(f"Write grid file: {grid_file_name} ")
+    grid_object.to_file(grid_file_name, fformat="egrid")
+    return grid_object
+
+
+def create_upscaled_grid():
```

Comment:
If we parametrize like I suggested in another comment, I think we could just have one function to create grids, as the only differences between `create_upscaled_grid` and `create_grid` seem to be `grid_file_name` and `grid_dimension`.

Reply:
Done